daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf_dev/daml_lf_1.proto

```diff
@@ -1337,6 +1363,25 @@ message DefDataType {
   Location location = 6;
 }
 
+// Type synonym definition
+message DefTypeSyn {
+
+  // name of the defined type synonym
+  oneof name {
+    DottedName name_dname = 1;
+
+    // *Must be a valid interned name*
+    int32 name_interned_dname = 8; // *Available in versions >= 1.7*
```

Comment:
Same here, you can remove the `name_dname` case. For consistency it is probably still nice fo have a wrapper but a `oneof` with only one entry should be sufficient for that.

Reply:
I tried this, but it causes type errors, which I don't want to deal with now.
So I didn't make this change.

Is it worth having a special case for names which must be interned, simply because they didn't exist prior to interning being implemented. Maybe! But if so, we can do this later.